autora/experimentalist/sampler/assumption.py

```diff
@@ -0,0 +1,69 @@
+from typing import Iterable
+
+import numpy as np
+from sklearn.metrics import mean_absolute_error as mae
+from sklearn.metrics import mean_squared_error as mse
+
+
+def assumption_sampler(X, y, model, n, loss=True, theorist=None):
+    """
+    Assumption Sampler challenges assumptions made by the Theorist.
+    It identifies points whose error are most dependent on the assumption made.
+    Assumptions take the form of hard-coding, which may be hyperparameters or arbitrarily chosen
+    sub-algorithms e.g. loss function
+    Because it samples with respect to a Theorist, this sampler cannot be used on the first cycle
+
+    Args:
+        X: pool of IV conditions to sample from
+        y: experimental results from most recent iteration
+        model: Scikit-learn model, must have `predict` method.
+        n: number of samples to select
+        loss: assumption to test: identify points that are most affected by choice of loss function
+        theorist: the Theorist, which employs the theory it has been hard-coded to demonstrate
+
+    Returns: Sampled pool
+
+    """
+
+    if isinstance(X, Iterable):
+        X = np.array(list(X))
+    current = None
+    if theorist:
+        pass  # add code to extract loss function from theorist object
+    idx = range(len(X))
+
+    if y is not None:
+        if loss:
+            if current is None:
+                current = mse
+                print(
+                    Warning(
+                        "Knowledge of Theorist Loss Function needed. MSE has been assumed."
+                    )
+                )
+            y_pred = model.predict(X)
+            print(y_pred.shape)
```

Comment:
We might want to avoid any print statements when regularly executing the code. Alternatively, you could include an additional boolean argument `verbose` in the function and then only print if verbose it set to true (the default would be false). E.g.,
```
if verbose:
   print(y_pred.shape)
```

Reply:
whoops! I didn't want that to be there at all. Thank you for catching that! I completely agree, we don't want print statements coming up normally. Since I deleted most of the print statements I don't see much use in keeping just this one in so I just removed it.